### PR TITLE
Feat: runnable examples in agent how to docs

### PR DESCRIPTION
**Pull request by bracesproul**
- [x] Conversational
- [x] OpenAI functions
- [ ] ~~Plan and Execute~~
- [x] ReAct
- [x] Structured tool chat
- [x] XML Agent
- [x] General LCEL example

### CODE REVIEW

File: docs/docs/modules/agents/agent_types/index.mdx [hunk 1]
```diff
@@ -30,6 +30,6 @@ This agent is designed to be used in conversational settings.
 The prompt is designed to make the agent helpful and conversational.
 It uses the ReAct framework to decide which tool to use, and uses memory to remember the previous conversation interactions.
 
-## [Plan-and-execute agents](/docs/modules/agents/agent_types/plan_and_execute)
+### [Plan-and-execute agents](/docs/modules/agents/agent_types/plan_and_execute)
```
Comment (jacoblee93) [inline docs/docs/modules/agents/agent_types/index.mdx:33]: Just de-emphasizing it?

File: docs/docs/modules/agents/agent_types/openai_functions_agent.mdx [hunk 2]
```diff
@@ -5,19 +5,32 @@ sidebar_position: 0
 
 # OpenAI functions
 
-Certain OpenAI models (like gpt-3.5-turbo-0613 and gpt-4-0613) have been fine-tuned to detect when a function should to be called and respond with the inputs that should be passed to the function.
+Certain OpenAI models (like `gpt-3.5-turbo-0613` and `gpt-4-0613`) have been fine-tuned to detect when a function should to be called and respond with the inputs that should be passed to the function.
```
Comment (jacoblee93) [inline docs/docs/modules/agents/agent_types/openai_functions_agent.mdx:8]: We can remove the `-0613` now

File: docs/docs/modules/agents/agent_types/plan_and_execute.mdx [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Plan and execute
 
+:::info Compatibility
```
Comment (jacoblee93) [inline docs/docs/modules/agents/agent_types/plan_and_execute.mdx:3]: Is this true?

File: docs/docs/modules/agents/agent_types/plan_and_execute.mdx [hunk 4]
```diff
@@ -17,6 +21,10 @@ This agent uses a two step process:
 The idea is that the planning step keeps the LLM more "on track" by breaking up a larger task into simpler subtasks.
 However, this method requires more individual LLM queries and has higher latency compared to Action Agents.
 
-**Note**: This agent currently only supports Chat Models.
+# With `PlanAndExecuteAgentExecutor`
+
+:::info
+This is an experimental chain and may not have all its kinks worked out yet.
```
Comment (jacoblee93) [inline docs/docs/modules/agents/agent_types/plan_and_execute.mdx:27]: `This is an experimental chain and is not recommended for production use yet.`

File: examples/src/agents/chat_convo_with_tracing_runnable.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import { pull } from "langchain/hub";
+import { PromptTemplate } from "langchain/prompts";
+import { RunnableSequence } from "langchain/schema/runnable";
+import { InputValues } from "langchain/schema";
+import { BufferMemory } from "langchain/memory";
+import { formatLogToString } from "langchain/agents/format_scratchpad/log";
+import { renderTextDescription } from "langchain/tools/render";
+import { ReActSingleInputOutputParser } from "langchain/agents/react/output_parser";
+
+process.env.LANGCHAIN_HANDLER = "langchain";
```
Comment (jacoblee93) [inline examples/src/agents/chat_convo_with_tracing_runnable.ts]: What's this for?

File: examples/src/agents/chat_convo_with_tracing_runnable.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+const tools = [
+  new SerpAPI(process.env.SERPAPI_API_KEY, {
+    location: "Austin,Texas,United States",
+    hl: "en",
+    gl: "us",
+  }),
+  new Calculator(),
+];
+/** Pull a prompt from LangChain Hub */
+const prompt = await pull<PromptTemplate>("hwchase17/react-chat");
```
Comment (jacoblee93) [inline examples/src/agents/chat_convo_with_tracing_runnable.ts:30]: Might be good to paste the pulled prompt in a comment if it's not too long

File: examples/src/agents/xml_runnable.ts [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+<observation>64 degrees</observation>
+
+When you are done, respond with a final answer between <final_answer></final_answer>. For example:
+
+<final_answer>The weather in SF is 64 degrees</final_answer>
+
+Begin!
+
+Question: {input}`;
+const createPrompt = () =>
```
Comment (jacoblee93) [inline examples/src/agents/xml_runnable.ts:49]: Do we really need this wrapper?

File: docs/docs/expression_language/cookbook/agents.mdx [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+<final_answer>The weather in SF is 64 degrees</final_answer>
+
+Begin!
+
+Question: {input}`;
+```
+
+```typescript
+const createPrompt = () =>
```
Comment (jacoblee93) [inline docs/docs/expression_language/cookbook/agents.mdx:82]: Do we really need this wrapper?

File: docs/docs/expression_language/cookbook/agents.mdx [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+  ChatPromptTemplate.fromMessages([
+    template,
+    AIMessagePromptTemplate.fromTemplate("{agent_scratchpad}"),
```
Comment (jacoblee93) [inline docs/docs/expression_language/cookbook/agents.mdx:85]: I would do:

```
  ChatPromptTemplate.fromMessages([
    ["human", template],
    ["ai", "{agent_scratchpad}"],
  ])
```

**Review by jacoblee93 — CHANGES_REQUESTED** [review]

**Comment by jacoblee93** [comment]
Make sure the prompts are correct and that all the messages are still the correct role `ai` vs. `system`

File: docs/docs/modules/agents/agent_types/index.mdx (hunk 1 above)
Comment (bracesproul) [reply docs/docs/modules/agents/agent_types/index.mdx:33]: there are two other items in this list, both with 3 #'s, just making it match

File: docs/docs/modules/agents/agent_types/plan_and_execute.mdx (hunk 3 above)
Comment (bracesproul) [reply docs/docs/modules/agents/agent_types/plan_and_execute.mdx:3]: Lemme double check, that change is just me reordering where it was on the page to match simliar docs

Comment (bracesproul) [reply docs/docs/modules/agents/agent_types/plan_and_execute.mdx:3]: Yes, looks like it is. It's calling `ChatAgent` a few times inside the code

File: examples/src/agents/xml_runnable.ts (hunk 7 above)
Comment (bracesproul) [reply examples/src/agents/xml_runnable.ts:49]: Nope, will remove

File: docs/docs/expression_language/cookbook/agents.mdx (hunk 8 above)
Comment (bracesproul) [reply docs/docs/expression_language/cookbook/agents.mdx:82]: Nope, will remove

**Review by jacoblee93 — APPROVED** [review]

**Comment by jacoblee93** [comment]
Thanks!

**Pull request merged**